### PR TITLE
Allow to access Nested array variables

### LINKED ISSUE #1009: Issue while accessing array item
```fortran

program example_lmder1

integer :: a = 10, y = 15
real :: x(3)
x = [1.0, 1.0, 1.0]

call foo()

contains

subroutine foo()
print *, a, y, x(1), x(2)
end subroutine

end program
```

gfortran output:
```
% gfortran a.f90 && ./a.out
          10          15   1.00000000       1.00000000
```

lfortran error:
```
handle_print(x);
  File "/Users/thebigbool/repos2/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5303
    this->visit_expr_wrapper(x.m_values[i], true);
  File "/Users/thebigbool/repos2/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 3831
    this->visit_expr(*x);
  File "../libasr/asr.h", line 4149
  File "../libasr/asr.h", line 3970
  File "/Users/thebigbool/repos2/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 1581
    LFORTRAN_ASSERT(llvm_symtab.find(v_h) != llvm_symtab.end());
AssertFailed: llvm_symtab.find(v_h) != llvm_symtab.end()
```

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes #1009 
Also related to https://github.com/certik/minpack/issues/11

### CODE REVIEW

**Comment by czgdp1807** [comment]
I will review it closely tomorrow. The solution we discussed for nested global arrays seems to work. Will merge this, and a bunch of other PRs and then start the sync tomorrow.

File: integration_tests/arrays_14.f90 [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+b = [14, 12]
+
+call foo()
+
+contains
+
+subroutine foo()
+if( a /= 10 ) error stop
+if( b(1) /= 14 ) error stop
+if( b(2) /= 12 ) error stop
```
Comment (czgdp1807) [inline integration_tests/arrays_14.f90:14]: What if I do, `b(1) = 15`? Will it be reflected in `b` after `call foo()`?

Comment (Smit-create) [reply integration_tests/arrays_14.f90:14]: Yes

**Comment by certik** [comment]
I rebased on top of the latest main. I think this is ok.

These nested variables should be later refactored to become some special "nested_variables" as part of the function signature, and then promoted to regular arguments, to avoid the hacks in the backends. Until then, this is ok.

**Pull request merged**